### DISCUSSION TITLE
How can I unzip the zip with the password

### DISCUSSION THREAD

**Discussion by ghost** [gildas-lormeau/zip.js#320] [Q&A]
I tried to unzip the zip with the password, but got this error.
<img width="797" alt="image" src="https://user-images.githubusercontent.com/57799539/161226704-bc3b7fae-aa7f-4f6a-822d-2eaa7e0c938c.png">
Is the location of my password field wrong?
this is my code.
```
  let passwd = "86edb94c-47fb-43f6-8e63-1b4ec7712151"
  const reader = new zip.ZipReader(new zip.BlobReader(data));
  const entries = await reader.getEntries();
  if (entries.length) {
    console.log('[ entries ] >', entries)
    const jsonDetail = await entries[499].getData(
      // writer
      new zip.TextWriter(), {
        password: passwd
      }
    );
    console.log(jsonDetail);
  }
  await reader.close();
```

**Comment by gildas-lormeau** [comment]
What version of zip.js are you using?

**Reply by ghost** [reply]
> 
"@zip.js/zip.js": "^2.4.7",

**Reply by gildas-lormeau** [reply]
Please update it to 2.4.9 and do the test again.

**Reply by ghost** [reply]
> Please update it to 2.4.9 and do the test again.

solved, thanks

**Reply by gildas-lormeau** [reply]
You're welcome!
